src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1146,58 +1095,83 @@ private static string UpdatePath(string basePath, string pathToAdd, ref int inse
+                // Trailing white spaces were already removed by 'StringSplitOptions.TrimEntries'.
+                addedPaths.Add(NormalizePath(p));
+            }
 
-                // Searching in effective 'result' value ensures that possible duplicates in pathsToAdd are handled correctly.
-                if (PathContainsSubstring(result.ToString(), subPathToAdd))
+            foreach (string subPathToAdd in newPaths)
+            {
+                // Remove the trailing directory separators.
+                // Trailing white spaces were already removed by 'StringSplitOptions.TrimEntries'.
+                string normalizedPath = NormalizePath(subPathToAdd);
+                if (addedPaths.Contains(normalizedPath))
                 {
-                    // The 'subPathToAdd' was already found - skip it.
+                    // The normalized sub path was already added - skip it.
                     continue;
                 }
 
-                // The 'subPathToAdd' was not found - add it.
+                // The normalized sub path was not found - add it.
                 if (insertPosition is -1 || insertPosition >= result.Length)
                 {
-                    // Append 'subPathToAdd' to the end.
-                    if (result.Length > 0 && result[^1] == Path.PathSeparator)
-                    {
-                        result.Append(subPathToAdd);
-                    }
-                    else
+                    // Append the normalized sub path to the end.
+                    if (result.Length > 0 && result[^1] != Path.PathSeparator)
                     {
-                        result.Append(Path.PathSeparator).Append(subPathToAdd);
+                        result.Append(Path.PathSeparator);
                     }
 
+                    result.Append(normalizedPath);
                     // Next insertion should happen at the end.
                     insertPosition = result.Length;
                 }
                 else
                 {
                     // Insert at the requested location.
                     // This is used by the user-specific module path, the shared module path (<Program Files> location), and the PSHome module path.
-                    string strToInsert = subPathToAdd + Path.PathSeparator;
+                    string strToInsert = normalizedPath + Path.PathSeparator;
                     result.Insert(insertPosition, strToInsert);
 
                     // Next insertion should happen after the just inserted string.
                     insertPosition += strToInsert.Length;
                 }
+
+                // Add it to the set.
+                addedPaths.Add(normalizedPath);
             }
 
             return result.ToString();
+
+            // Local function to normalize a path by removing trailing directory separators.
+            static string NormalizePath(string path)
+            {
+                string normalizedPath = path.TrimEnd(Path.DirectorySeparatorChar);
```

Comment:
Why not use NormalizePath method from FileSystemProvider? Then we will follow the basic PowerShell behavior regarding path separators.
Otherwise, we could use Path.TrimEndingDirectorySeparator()